pkg/grpc/server.go

```diff
@@ -61,3 +77,83 @@ func (s *server) StartNonBlocking() error {
 	}()
 	return nil
 }
+
+func (s *server) stop() {
+	s.srv.GracefulStop()
+}
+
+func (s *server) getGRPCServer() (*grpc_go.Server, error) {
+	opts := []grpc_go.ServerOption{}
+
+	if s.tracingSpec.Enabled {
+		opts = append(opts, grpc_go.StreamInterceptor(diag.TracingGRPCMiddleware(s.tracingSpec)), grpc_go.UnaryInterceptor(diag.TracingGRPCMiddlewareUnary(s.tracingSpec)))
+	}
+
+	if s.authenticator != nil {
+		log.Info("sending workload csr request to sentry")
+		signedCert, err := s.authenticator.CreateSignedWorkloadCert(s.config.DaprID)
+		if err != nil {
+			return nil, fmt.Errorf("error from authenticator CreateSignedWorkloadCert: %s", err)
+		}
+
+		tlsCert, err := tls.X509KeyPair(signedCert.WorkloadCert, signedCert.PrivateKeyPem)
+		if err != nil {
+			return nil, fmt.Errorf("error creating x509 Key Pair: %s", err)
+		}
+
+		caCertPool := signedCert.TrustChain
+		tlsConfig := tls.Config{
+			Certificates: []tls.Certificate{tlsCert},
```

Comment:
@yaron2 I missed this part in the review. I realized that we do not have fallback for old cert in the meeting. Why not allowing the old cert ? it prevents unauthorization error (401 or 403) when callee has new cert and caller has old cert. retry behavior by unauthorization response is really heavy and bottleneck in the real production. 

cc/ @amanbha @artursouza

Reply:
As discussed in person, this is a non issue due to the fact the same private key is used to verify the issuer cert that signs the workload certs.